File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    let m: { [key: number]: number } = {};

    for (let i = 0; i < nums.length; i++) {
      let complment = target - nums[i];
      if (complment in m) return [m[complment], i];
      m[nums[i]] = i;
    }

    return [];
};

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
      Dictionary<int, int> m = new Dictionary<int, int>();
      for (int i = 0; i < nums.Length; ++i)
      {
          if (m.ContainsKey(target - nums[i]))
          {
              return new int[] { m[target - nums[i]], i };
          }
          m[nums[i]] = i;
      }
      return new int[] { };
    }
}